### PR TITLE
Open version history in new window

### PR DESCRIPTION
**Pull request by VladMartyshevskyi**
Open version history in new window using PostMessage.

### CODE REVIEW

File: services/src/main/java/org/exoplatform/officeonline/WOPIService.java [hunk 1]
```diff
@@ -80,6 +80,7 @@
 import org.exoplatform.webui.application.WebuiRequestContext;
 import org.exoplatform.webui.application.portlet.PortletRequestContext;
 
+// TODO: Auto-generated Javadoc
```
Comment (pnedonosko) [inline services/src/main/java/org/exoplatform/officeonline/WOPIService.java]: Cleanup?

File: services/src/main/java/org/exoplatform/officeonline/WOPIService.java [hunk 2]
```diff
@@ -1121,11 +1124,11 @@ protected void addFileURLProperties(Map<String, Serializable> map, Node node, Ed
...
-    String explorerLink = symlink != null ? explorerLink(symlink.getPath()) : explorerLink(node.getPath());
-    URI explorerUri = explorerUri(config.getBaseUrl(), explorerLink);
-    if (explorerUri != null) {
-      map.put(CLOSE_URL, explorerUri.toString());
-      map.put(FILE_VERSION_URL, explorerUri.toString() + "&versions=true");
+    Node explorerNode = symlink != null ? symlink : node;
+    String explorerUrl = getExplorerURL(explorerNode, config.getBaseUrl());
+    if (explorerUrl != null) {
+      map.put(CLOSE_URL, explorerUrl);
+      map.put(FILE_VERSION_URL, explorerUrl + "&versions=true");
```
Comment (pnedonosko) [inline services/src/main/java/org/exoplatform/officeonline/WOPIService.java]: StringBuilder will be more efficient if it's often called op.

File: webapp/src/main/java/org/exoplatform/officeonline/portlet/EditorPortlet.java [hunk 3]
```diff
@@ -100,7 +100,9 @@ public void view(RenderRequest request, RenderResponse response) throws IOExcept
         if (validAction(node, action)) {
           String actionURL = wopiService.getActionUrl(requestInfo, fileId, null, action);
           String filename = node.getName();
-          callModule("officeonline.initEditor(" + token.toJSON() + ", \"" + actionURL + "\", \"" + filename + "\");");
+          String explorerUrl = wopiService.getExplorerURL(node, config.getBaseUrl());
+          String versionsUrl = explorerUrl + "&versions=true";
```
Comment (pnedonosko) [inline webapp/src/main/java/org/exoplatform/officeonline/portlet/EditorPortlet.java]: StringBuilder is better from perf PoV

File: webapp/src/main/webapp/js/officeonline.js [hunk 4]
```diff
@@ -288,6 +288,7 @@
     // Constants:
     var DOCUMENT_SAVED = "DOCUMENT_SAVED";
     var FILE_RENAME = "File_Rename";
+    var FILE_VERSIONS = "UI_FileVersions";
```
Comment (pnedonosko) [inline webapp/src/main/webapp/js/officeonline.js]: We can use ES6 `const` to mark our constants be really ones. Platform 6 requires ES6 compliant browser only.

**Review by pnedonosko — CHANGES_REQUESTED** [review]

File: webapp/src/main/webapp/js/officeonline.js (hunk 4 above)
Comment (VladMartyshevskyi) [reply webapp/src/main/webapp/js/officeonline.js]: Fixed: https://github.com/exo-addons/office-online/pull/28/commits/09dd43c4c6497a4cd87cc727a522b7c45c3eefdd

File: webapp/src/main/java/org/exoplatform/officeonline/portlet/EditorPortlet.java (hunk 3 above)
Comment (VladMartyshevskyi) [reply webapp/src/main/java/org/exoplatform/officeonline/portlet/EditorPortlet.java]: Fixed: https://github.com/exo-addons/office-online/pull/28/commits/09dd43c4c6497a4cd87cc727a522b7c45c3eefdd

File: services/src/main/java/org/exoplatform/officeonline/WOPIService.java (hunk 2 above)
Comment (VladMartyshevskyi) [reply services/src/main/java/org/exoplatform/officeonline/WOPIService.java]: Fixed: https://github.com/exo-addons/office-online/pull/28/commits/09dd43c4c6497a4cd87cc727a522b7c45c3eefdd

File: services/src/main/java/org/exoplatform/officeonline/WOPIService.java (hunk 1 above)
Comment (VladMartyshevskyi) [reply services/src/main/java/org/exoplatform/officeonline/WOPIService.java]: Cleanup: https://github.com/exo-addons/office-online/pull/28/commits/09dd43c4c6497a4cd87cc727a522b7c45c3eefdd

**Review by pnedonosko — APPROVED** [review]

**Pull request merged**